core-bundle/src/Resources/contao/classes/DataContainer.php

```diff
@@ -1179,8 +1179,16 @@ protected function panel()
 			{
 				$panel = '';
 
+				// The limit menu depends on other panels that may set a filter query, e.g. search and filter.
+				// In order to correctly calculate the total row count, the limit menu must be compiled last.
+				// We insert a placeholder here and compile the limit menu after all other panels.
+				if ($strSubPanel == 'limit')
+				{
+					// Set placeholder to inject compiled limit menu later
+					$panel = '###limit_menu###';
+				}
 				// Regular panels
-				if ($strSubPanel == 'search' || $strSubPanel == 'limit' || $strSubPanel == 'sort')
+				elseif ($strSubPanel == 'search' || $strSubPanel == 'sort')
```

Comment:
Now that nearly all `$strSubPanel` variants are handled individually, we could maybe replace the last two `$panel = $this->{$strSubPanel . 'Menu'}(…);` expressions with real function calls? Took me a while to get what this was doing.

Reply:
Yeah, I agree this takes a moment to figure that out. Would you continue with if {} elseif {} in this case?

Reply:
Yes, or a `switch` statement. But let's hear what others think about this first.

Reply:
Doesn't sound like something related to this PR (fixing the limit menu) 😇

Reply:
What's wrong with `$this->{$strSubPanel . 'Menu'}()`? It is a real function call, isn't it?

Reply:
I refactored this part to a switch statement. Definitely helps with readability.

Reply:
> What's wrong with $this->{$strSubPanel . 'Menu'}()? It is a real function call, isn't it? smile

:smile: I was more referring to things like this where the dynamic part is unnecessary:

```php
elseif ($strSubPanel == 'filter')
{
    $panel = $this->{$strSubPanel . 'Menu'}(++$intFilterPanel);
}
```
&rarr;
```php
elseif ($strSubPanel == 'filter')
{
    $panel = $this->filterMenu(++$intFilterPanel);
}
```